docs/streams/upgrade-guide.html

```diff
@@ -72,6 +72,42 @@ <h1>Upgrade Guide and API Changes</h1>
         More details about the new config <code>StreamsConfig#TOPOLOGY_OPTIMIZATION</code> can be found in <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-295%3A+Add+Streams+Configuration+Allowing+for+Optional+Topology+Optimization">KIP-295</a>.
     </p>
 
+    <h3><a id="streams_api_changes_240" href="#streams_api_changes_240">Streams API changes in 2.4.0</a></h3>
+
+    <!-- Placeholder KIP-213 -->
+
+    <!-- Placeholder KIP-307 -->
+    <!-- Placeholder KIP-479 -->
+
+    <!-- Placeholder KIP-429 -->
+
+    <!-- Placeholder KIP-444 -->
+    <!-- Placeholder KIP-471 -->
+
+    <p>
+        Kafka Streams <code>test-utils</code> got improved via
+        <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-470%3A+TopologyTestDriver+test+input+and+output+usability+improvements">KIP-470</a>
+        to simplify testing with the <code>TopologyTestDriver</code>.
+        We deprecated <code>ConsumerRecordFactory</code>, <code>TopologyTestDriver#pipeInput()</code>,
+        <code>OutputVerifier</code>, as well as <code>TopologyTestDriver#readOutput()</code> and replace them with
+        <code>TestInputTopic</code> and <code>TestOutputTopic</code>, respectively.
+        We also introduced a new class <code>TestRecord</code> that simplifies assertion code.
+        Additionally, a few minor API improvements were applied.
+    </p>
+
+    <p>
+        In 2.3.0, we deprecated <code>WindowStore#put(K key, V value)</code> that should never be used.
+        Instead already existing <code>WindowStore#put(K key, V value, long windowStartTimestamp)</code> should be used
+        (<a href="https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=115526545">KIP-474</a>).
+    </p>
+
+    <p>
+        Furthermore, the <code>PartitionGrouper</code> interface and it's corresponding configuration parameter
```

Comment:
Maybe explain why, and how users who already leverage on this should expect what kind of impact and how to change their code accordingly.

Reply:
Not sure what guidance we can give? We remove the interface without replacement, hence, user can't do anything about it, but stop using the feature (if if they really need it -- what I actually doubt -- they would need to stay on an older version). Happy to add anything else if you have suggestions?

Reply:
+1 agree with Matthias here (also, ultra small/pedantic nit: it's -> its)

Reply:
I think we should at least state that if you are already using the PartitionGrouper to group multiple partitions into a single task, you should now get more tasks and hence should consider reset your applications.

Reply:
I guess a custom `PartitionGrouper` could also be use to split tasks, and hence, after removing it, there might be fewer task. Will add a comment that tasks might change and hence, a reset would be necessary to move forward.